RecoJets/JetProducers/src/JetSpecific.cc

```diff
@@ -256,55 +260,56 @@ bool reco::makeSpecific(std::vector<reco::CandidatePtr> const& particles, PFJet:
     const Candidate* pfCand = itParticle->get();
     if (pfCand) {
       const PFCandidate* pfCandCast = dynamic_cast<const PFCandidate*>(pfCand);
+      float weight = (weights != nullptr) ? (*weights)[*itParticle] : 1.0;
       if (pfCandCast)
-        HOEnergy += pfCandCast->hoEnergy();
+        HOEnergy += pfCandCast->hoEnergy() * weight;
```

Comment:
I was puzzled by somewhat significant change in this variable in the puppi jets after this PR.
Apparently, the treatment here is different in this PR compared to the baseline.
In the baseline the PUPPI PFCandidate ```hoEnergy()``` is the same as its parent, because PuppiProducer changes only the momentum (via ```.setP4```), which does not affect PF-specific quantities like matching calo.

I can see how this PR would give a more consistent treatment. So, it's fine if this is in fact intended.
Please add a note in the PR description of this difference.